lib/helpers.bash

```diff
@@ -224,6 +224,9 @@ _bash-it-update-() {
 
   cd "${BASH_IT}" || return
 
+  DIFF=$(git diff --name-status)
+  [ -n "$DIFF" ] && echo -e "Either commit your changes using git commit, or clean them...\n$DIFF" && return 1
+
```

Comment:
Ok, code-wise this seems great!

A couple things just occurred to me regarding the error message:

* We probably need to start the message with something like `"Local changes detected"`
* I don't know if its a good idea to suggest that the user commit local changes.  It might better to suggest they `stash` them instead.

Reply:
@davidpfarrell Already discussed here: https://github.com/Bash-it/bash-it/issues/1889#issuecomment-859177548

Reply:
@NoahGorny  ^^

Reply:
How about "Local changes detected, unable to automatically update. Either commit your changes (git commit), stash them (git stash), or discard them (git checkout -- .)

Let me know what you think!

Reply:
I am not a fan of encouraging users to commit their local changes, as this will cause _incoming_ conflicts which can prevent updates from succeeding in the future.

I would much prefer to focus on stashing changes so the user would get post-update conflicts on stash pop, but this wouldn't prevent the update from succeeding.

Keeping local changes un-committed also helps keep the user *aware* that there are indeed local changes.

Reply:
@NoahGorny Yes, also we can give them option to select like `[dD/sS/cC/aA]`, Where `aA` means abort operation

Reply:
@NoahGorny @davidpfarrell Let's not overcomplicate a message. We can put a generic message

"Local changes detected in bash-it directory. Clean $BASH_IT directory to proceed"

Let me know if this works

Reply:
Done